flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClientTest.java

```diff
@@ -630,4 +630,37 @@ private KubernetesConfigMap buildTestingConfigMap() {
                         .withData(data)
                         .build());
     }
+
+    @Test
+    void testMockPrePreparedResources() throws Exception {
+        // regenerate a test JM spec
+        Deployment testDeployment = kubernetesJobManagerSpecification.getDeployment();
+        List<HasMetadata> testAccompanyingResources =
+                kubernetesJobManagerSpecification.getAccompanyingResources();
+        List<HasMetadata> mockPrePreparedResources = new ArrayList<>();
+        for (int i = 0; i < 3; i++) {
+            Pod mockPod =
+                    new PodBuilder()
+                            .editOrNewMetadata()
+                            .withName(String.format("mock-pod-preprepared-resource%d", i))
+                            .endMetadata()
+                            .editOrNewSpec()
+                            .endSpec()
+                            .build();
+            mockPrePreparedResources.add(mockPod);
+        }
+        KubernetesJobManagerSpecification testKubernetesJobManagerSpecification =
+                new KubernetesJobManagerSpecification(
+                        testDeployment, testAccompanyingResources, mockPrePreparedResources);
+        flinkKubeClient.createJobManagerComponent(testKubernetesJobManagerSpecification);
+
+        // check the preprepared resources had been created.
+        final List<Pod> resultPods = kubeClient.pods().inNamespace(NAMESPACE).list().getItems();
+        assertThat(resultPods).hasSize(3);
```

Comment:
Shouldn't this be at least 4 pods ?
1 pod for the JM itself and 3 `mock-pod-preprepared-resource-**`.

Reply:
This is UT, it won't create any real Pods, just a k8s models objects. Actually, JM will be created by K8S deployment, the real env, that's true it should be 4 pods.